Tests/Shared.Specs/NullableNumericAssertionSpecs.cs

```diff
@@ -1243,6 +1243,65 @@ public void When_asserting_not_approximately_and_nullable_decimal_is_approximati
 
         #endregion
 
+        #region Match
+
+
+        [Fact]
+        public void When_nullable_value_satisfies_predicate_it_should_not_throw()
+        {
+            //-----------------------------------------------------------------------------------------------------------
+            // Arrange
+            //-----------------------------------------------------------------------------------------------------------
+            int? nullableInteger = 1;
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Act / Assert
+            //-----------------------------------------------------------------------------------------------------------
+            nullableInteger.Should().Match(o => o.HasValue);
+        }
+
+        [Fact]
+        public void When_nullable_value_does_not_match_the_predicate_it_should_throw()
+        {
+            //-----------------------------------------------------------------------------------------------------------
+            // Arrange
+            //-----------------------------------------------------------------------------------------------------------
+            int? nullableInteger = 1;
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Act
+            //-----------------------------------------------------------------------------------------------------------
+            Action act = () => nullableInteger.Should().Match(o => !o.HasValue, "it is not initialized yet");
```

Comment:
Please include `becauseArgs` to verify they are properly passed along to `BecauseOf`.
E.g.
```c#
Match(o => !o.HasValue, "because we want to test the failure {0}", "message");
...
WithMessage("Expected value to match Not(o.HasValue) because we want to test the failure message.")
```